05_syntax_lab/06.py

```diff
@@ -1,9 +1,15 @@
-"""
-Write a program that randomizes 2 numbers
-and calculates their least common multiplier,
-that is the smallest number that is divisable
-by both.
-For example if the numbers were 4 and 6,
-program should print 12
-"""
-
+################### Part 6
+r1 = randint(1, 10)
+r2 = randint(1, 10)
+maximum = r1 * r2
+print('r1 = %d r2 = %d max quotient = %d' % (r1, r2, maximum))
+i = 1
+min_quotient = []
+while i <= maximum:
+    if (i % r1 == 0) and (i % r2 == 0):
+        min_quotient.append(i)
+        #print('{}'.format(min_quotient))
+    i += 1
+
+print('%d' % min(min_quotient))
+

```

Comment:
Maybe this one:

from random import randint

r1 = randint(1, 10)
r2 = randint(1, 10)

print('%d  %d' % (r1, r2))


mini = min(r1,r2)
maxi = max(r1,r2)
i = int(maxi/mini)

while True:
    if i*mini % maxi == 0:
        break
    i += 1

print('The least common multiple of the two numbers is: {}'.format(i*mini))


From: Gabor Szabo [mailto:notifications@github.com]
Sent: Tuesday, July 05, 2016 6:25 PM
To: ynonp/python-examples-verint-2016-07
Cc: Rosenthal, Hanan (Embedded); Author
Subject: Re: [ynonp/python-examples-verint-2016-07] ToCode Solution: תרגול תחביר (#14)


In 05_syntax_lab/06.py<https://github.com/ynonp/python-examples-verint-2016-07/pull/14#discussion_r69582892>:

> -

> +################### Part 6

> +r1 = randint(1, 10)

> +r2 = randint(1, 10)

> +maximum = r1 * r2

> +print('r1 = %d r2 = %d max quotient = %d' % (r1, r2, maximum))

> +i = 1

> +min_quotient = []

> +while i <= maximum:

> +    if (i % r1 == 0) and (i % r2 == 0):

> +        min_quotient.append(i)

> +        #print('{}'.format(min_quotient))

> +    i += 1

> +

> +print('%d' % min(min_quotient))

> +

This algorithm will be quite slow if the numbers are big (not just 10*9). Could you find and implement a faster one?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ynonp/python-examples-verint-2016-07/pull/14/files/35968e51a6505b2ba16b664b20eb65939b37648b..01d35d1fa5d08b5065c3ff1679607de44b021e1e#r69582892>, or mute the thread<https://github.com/notifications/unsubscribe/ATVHXhCADjuSt0tFytLcyzV-sQe6ODRKks5qSnczgaJpZM4JE3pq>.

-- 
This electronic message may contain proprietary and confidential 
information of Verint Systems Inc., its affiliates and/or subsidiaries.
The information is intended to be for the use of the individual(s) or
entity(ies) named above. If you are not the intended recipient (or 
authorized to receive this e-mail for the intended recipient), you may not 
use, copy, disclose or distribute to anyone this message or any information 
contained in this message. If you have received this electronic message in 
error, please notify us by replying to this e-mail.